src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -136,14 +136,32 @@ export function start(manifest) {
 
 The following methods are provided:
 
-##### `app.render(request)`
+##### `app.render(request, routeData, locals)`
 
-This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes, that do not render pages.
+This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes that do not render pages.
 
 ```js
 const response = await app.render(request);
 ```
 
+The method accepts a mandatory `request` argument, and two other optional arguments: [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/guides/middleware/#locals).
+
+Use the first argument when you already know the route to render. Doing so will bypass the internal call to [`app.match`](#appmatchrequest) to determine the route to render.
```

Comment:
```suggestion
The method accepts a mandatory `request` argument, and two other optional arguments: [`routeData`](/en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/guides/middleware/#locals).

Provide a value for `routeData` if you already know the route to render. Doing so will bypass the internal call to [`app.match`](#appmatchrequest) to determine the route to render.
```
I'm not sure whether the above is a correct reframing, but the example below *uses* `routeData` (i.e. provides a value for it, even if it's `undefined` vs not). So, I don't think the word "uses" is exactly what we're going for here?

Also question: *if* (and I have no idea whether this is true) `routeData` and `locals` must ALWAYS be specified together, or even whether if you want to specify locals, you NEED *something* for the middle argument (e.g. `undefined` in this case), then this would be the place to state that explicitly.

Reply:
> I'm not sure whether the above is a correct reframing, but the example below *uses* `routeData` (i.e. provides a value for it, even if it's `undefined` vs not). So, I don't think the word "uses" is exactly what we're going for here?
> 



> Also question: *if* (and I have no idea whether this is true) `routeData` and `locals` must ALWAYS be specified together, or even whether if you want to specify locals, you NEED *something* for the middle argument (e.g. `undefined` in this case), then this would be the place to state that explicitly.


The second and third parameters are optionals, but if you need to pass the third, you can't skip the second, and you pass `undefined`. Internally the function for the existence of `routeData` and if it's `undefined`, internally the function will use `.match`.